### PR TITLE
[SIG-3211] Enable typescript

### PR DESCRIPTION
**Pull request by MauritsR**
Adds Typescript configuration (changes by @aditudorache, see https://github.com/Amsterdam/signals-frontend/pull/1188)

### CODE REVIEW

**Comment by MauritsR** [comment]
- [ ] fix tests

File: src/signals/settings/users/Overview/__tests__/Overview.test.js [hunk 1]
```diff
@@ -335,8 +335,8 @@ describe('signals/settings/users/containers/Overview', () => {
 
     expect(dispatch).not.toHaveBeenCalled();
 
-    const filterByUserName = await findByTestId('filterUsersByUsername');
-    const filterByUserNameInput = filterByUserName.querySelector('input');
+    let filterByUserName = await findByTestId('filterUsersByUsername');
```
Comment (boris-arkenaar) [inline src/signals/settings/users/Overview/__tests__/Overview.test.js:338]: Prefor to use separate tests instead of `let`s, `unmount`s and `rerender`s.

Comment (aditudorache) [reply src/signals/settings/users/Overview/__tests__/Overview.test.js:338]: We generally don't use `let` in the code. The only exception is in some test scenarios where the initialization is done in the `beforeEach` function.

**Comment by MauritsR** [comment]
Closed in favor of https://github.com/Amsterdam/signals-frontend/pull/1320

**Pull request closed without merging**